lib/std/tempfiles.nim

```diff
@@ -94,22 +94,26 @@ proc getTempDirImpl(dir: string): string {.inline.} =
   result = dir
   if result.len == 0:
     result = getTempDir()
+  elif not dirExists(result):
+    raise newException(IOError, "the specified directory does not exist: " & dir)
 
 proc genTempPath*(prefix, suffix: string, dir = ""): string =
```

Comment:
What kind of API is that anyway... "prefix" and "suffix"? What if "prefix" contains a slash? Strings suck.

Reply:
I would instead make an in-place func kinda like:

```nim
var dir = "/tmp"  # must exist
createTempDir(dir)
#  dir == "/tmp/3j4k2l3n"
```